### PR TITLE
core: check excessBlobGas in front

### PR DESCRIPTION
**Pull request by jsvisa**
We should check ExcessBlobGas before `buyGas`, or else an nil pointer error is occurred

### CODE REVIEW

**Comment by karalabe** [comment]
Myeah, that check doesn't make any sense at all. It's crashing anyway, it's a programming error. Just remove it altogether.

**Comment by jsvisa** [comment]
> Myeah, that check doesn't make any sense at all. It's crashing anyway, it's a programming error. Just remove it altogether.

Yes, for the result it also crashed, but it crashed for a readable message(`missing field excess data gas`) instead of nil pointer call stack. BTW, does that never happen if a valid blob tx, if so I think we can remove them all.

File: core/state_transition.go [hunk 1]
```diff
@@ -333,6 +329,7 @@ func (st *StateTransition) preCheck() error {
 	if st.evm.ChainConfig().IsCancun(st.evm.Context.BlockNumber, st.evm.Context.Time) {
 		if st.blobGasUsed() > 0 {
 			// Check that the user is paying at least the current blob fee
+			// ExcessBlobGas must not be nil, which is checked in eip4844.VerifyEIP4844Header
```
Comment (karalabe) [inline core/state_transition.go:332]: No need for this log. The code would get very very full of these if we were to add all kinds of notes on why one field or another cannot be nil.

**Comment by jsvisa** [comment]
@karalabe PTAL, thanks

**Review by karalabe — APPROVED** [review]
LGTM, thanks

**Pull request merged**